openlibrary/views/loanstats.py

```diff
@@ -96,3 +98,47 @@ def is_enabled(self):
     def GET(self):
         stats = WLStats()
         return app.render_template("stats/waitinglists.html", stats)
+
+class readinglog_stats(app.view):
+    path = "/stats/readinglog"
+
+    def GET(self):
+        ONE_MONTH_DATE = datetime.date.today() - datetime.timedelta(days=28)
+        MAX_LEADERBOARD_SIZE = 50
+        i = web.input(limit="10")
+        limit = int(i.limit) if int(i.limit) < 51 else 50
+
+        def readinglog_stats(limit=limit):
+            if 'env' not in web.ctx:
+                delegate.fakeload()
+            most_read = Bookshelves.most_logged_books(
+                Bookshelves.PRESET_BOOKSHELVES['Already Read'], limit=limit)
+            most_wanted = Bookshelves.most_logged_books(                
+                Bookshelves.PRESET_BOOKSHELVES['Want to Read'], limit=limit)
+
+            return {
+                'total_books_logged': {
+                    'total': Bookshelves.total_books_logged(),
+                    'month': Bookshelves.total_books_logged(since=ONE_MONTH_DATE)
+                },
+                'total_users_logged': {
+                    'total': Bookshelves.total_unique_users(),
+                    'month': Bookshelves.total_unique_users(since=ONE_MONTH_DATE)
+                },
+                'leaderboard': {
+                    'most_read': most_read,
+                    'most_wanted': most_wanted
+                }
+            }
+        stats =  cache.memcache_memoize(
+            readinglog_stats, 'stats.readlinglog_stats', timeout=cache.HOUR)(limit)
+
+        # Fetch works from solr and inject into leaderboard
```

Comment:
What's the reliability of these stats, given they're from solr?

Reply:
@labanlabs the readinglog stats are going directly from postgres so they are accurate. They are tracked in a different way than most of our systems (intentionally). That said, the results are cached for an hour, so we expect up to an hour delay in the results.